### PR TITLE
add missing variable fields like defaultValue and isDefaulted

### PR DESCRIPTION
**Pull request by ajwootto**
- add missing fields to variable like type, defaultvalue and isDefaulted
- remove extraneous id field

### CODE REVIEW

**Comment by ajwootto** [comment]
Current dependencies on/for this PR:
* main
  * **PR #39** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/39" target="_blank">Graphite</a> 
    * **PR #40** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/40" target="_blank">Graphite</a> 
      * **PR #41** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/41" target="_blank">Graphite</a>  👈
        * **PR #42** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/42" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/41?utm_source=stack-comment).

File: DevCycle.SDK.Server.Cloud.MSTests/TestResponse.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public static Variable<bool> GetVariableByKeyAsync()
             {
                 {
                     "test-false",
-                    new Variable<bool>(Guid.NewGuid().ToString(), "test-false", TypeEnum.Boolean, false)
```
Comment (JamieSinn) [inline DevCycle.SDK.Server.Cloud.MSTests/TestResponse.cs:33]: why remove the random id?

File: DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs [hunk 2]
```diff
@@ -47,15 +42,22 @@ public Variable(string id = default, string key = default, TypeEnum type = defau
...
+            } ;
+
             Value = value;
 
             IsDefaulted = false;
         }
 
         public Variable(string key, T defaultValue, string evalReason)
         {
+            Console.WriteLine("MAKING DEFAULT VARIABLE");
```
Comment (elliotCamblor) [inline DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs]: log

File: DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs [hunk 3]
```diff
@@ -47,15 +42,22 @@ public Variable(string id = default, string key = default, TypeEnum type = defau
                 throw new InvalidDataException("value is a required property for Variable and cannot be null");
             }
 
+            if (defaultValue == null)
```
Comment (JamieSinn) [inline DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs:45]: you don't set defaultvalue if it's passed in

File: DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs (hunk 2 above)
Comment (JamieSinn) [inline DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs]: LOUD NOISES

File: DevCycle.SDK.Server.Cloud/Api/DVCCloudClient.cs [hunk 4]
```diff
@@ -86,6 +86,8 @@ public async Task<Variable<T>> VariableAsync<T>(User user, string key, T default
             try
             {
                 variable = await GetResponseAsync<Variable<T>>(user, urlFragment, queryParams);
+                variable.DefaultValue = defaultValue;
+                variable.IsDefaulted = false;
```
Comment (JamieSinn) [inline DevCycle.SDK.Server.Cloud/Api/DVCCloudClient.cs:90]: How do you know whether or not it was defaulted?

File: DevCycle.SDK.Server.Cloud.MSTests/TestResponse.cs (hunk 1 above)
Comment (ajwootto) [reply DevCycle.SDK.Server.Cloud.MSTests/TestResponse.cs:33]: its not part of the variable class as defined in Node (which is our reference)

File: DevCycle.SDK.Server.Cloud/Api/DVCCloudClient.cs (hunk 4 above)
Comment (ajwootto) [reply DevCycle.SDK.Server.Cloud/Api/DVCCloudClient.cs:90]: cuz if its defaulted then `GetResponseAsync` will throw (because the request is 404)

**Review by JamieSinn — APPROVED** [review]

File: DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs [hunk 5]
```diff
@@ -64,28 +67,26 @@ public Variable(string key, T defaultValue, string evalReason)
...
         public string Key { get; set; }
 
 
         /// <summary>
         /// Variable value can be a string, number, boolean, or JSON
         /// </summary>
         /// <value>Variable value can be a string, number, boolean, or JSON</value>
-        [DataMember(Name="value", EmitDefaultValue=false)]
+        [DataMember(Name="value")]
         public T Value { get; set; }
```
Comment (JamieSinn) [inline DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs:84]: Now that I think of it - `T` will actually become `JObject` or `JArray` - which we should check for how to handle JSON types.

Comment (ajwootto) [reply DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs:84]: i think it parses it as the type that is expected given the type T from the default value

Comment (JamieSinn) [reply DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs:84]: I'm more thinking from the JSON object representation - which might mess up the `Type` prop. 

Though - it's just something we'd have to test on the example app with a JSON variable.

Comment (ajwootto) [reply DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs:84]: default values also can't be arrays, that's not a valid type for a variable

Comment (JamieSinn) [reply DevCycle.SDK.Server.Common/Model/Cloud/Variable.cs:84]: It is now that you've set the default value to `T`. `JObject` is from Newtonsoft - which is how JSON variables get deserialized into C# - which is gonna add a bit of confusion on type enums.

**Pull request merged**